### PR TITLE
[DOCS] Adds anomaly detection resources and advanced concepts sections

### PR DESCRIPTION
**Pull request by lcawl**
Related to https://github.com/elastic/stack-docs/pull/1726

Requires https://github.com/elastic/stack-docs/pull/1731

### Screenshots

image

### CODE REVIEW

File: docs/en/stack/ml/anomaly-detection/index.asciidoc
Comment (szabosteve) [inline docs/en/stack/ml/anomaly-detection/index.asciidoc:37]: Would it be worth it to still keep these files in the ES repo? Is it feasible to move them in the stack-docs repo in a separate PR after the reorg is done? So all of the AD-related conceptual docs files would be in the same place.

**Review by szabosteve — APPROVED** [review]
I have only one question, otherwise LGTM after the appendices are fixed! Thanks!

**Comment by lcawl** [comment]
@elasticmachine run elasticsearch-ci/docs

**Comment by lcawl** [comment]
@elasticmachine run elasticsearch-ci/docs

**Comment by lcawl** [comment]
@elasticmachine update branch

File: docs/en/stack/ml/anomaly-detection/index.asciidoc
Comment (lcawl) [reply docs/en/stack/ml/anomaly-detection/index.asciidoc:37]: It's definitely something to consider, but I'll defer that decision for now.

**Pull request merged**
